.github/workflows/postman-collection-runner.yml

```
@@ -0,0 +1,103 @@
+name: Run postman tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - "releases/*"
+  pull_request:
+  merge_group:
+    types: [checks_requested]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  DATABASE_URL: postgres://db_user:db_pass@localhost/encryption_db
+  USER: db_user
+  PASSWORD: db_pass
+  DB_NAME: encryption_db
+
+jobs:
+  runner:
+    name: Run postman tests
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: "postgres:14.5"
+        env:
+          POSTGRES_USER: db_user
+          POSTGRES_PASSWORD: db_pass
+          POSTGRES_DB: encryption_db
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v4
+
+      - name: Install Rust
+        if: ${{ ((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) || (github.event_name == 'merge_group')}}
```

this can be set as an env at `L22` and use it down the line? that way, it will be more readable:

something like: 

```yml
env:
  RUN_TESTS: ${{ ((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) || (github.event_name == 'merge_group')}}
```

```suggestion
        if: ${{ env.RUN_TESTS == 'true' }}
```

also, it is better to have a step soon after repository checkout:

```yaml
      - name: Skip tests for PRs from forks
        shell: bash
        if: ${{ env.RUN_TESTS == 'false' }}
        run: echo 'Skipping tests for PRs from forks'
```